src/dachshund/scorer.rs

```diff
@@ -89,17 +86,12 @@ impl Scorer {
         &self,
         candidate: &Candidate<TGraph>,
     ) -> CLQResult<f32> {
-        // non_core_counts[0] currently corresponds to core nodes
-        let mut non_core_counts: Vec<usize> = vec![0; self.num_non_core_types + 1];
-        for &non_core_id in &candidate.non_core_ids {
-            let non_core_id = candidate
-                .get_node(non_core_id)
-                .non_core_type
-                .ok_or_else(CLQError::err_none)?;
-            non_core_counts[non_core_id.value()] += 1;
-        }
-        let mut score: f32 = 0.0;
-        for non_core_count in non_core_counts {
+        let non_core_counts = candidate.get_non_core_counts();
+        // Starts with a base of ln(1) for each missing type
+        // (including core nodes to mimic a quirk of the previous implementation.)
+        let mut score: f32 = (1.0 as f32).ln();
+        score *= (self.num_non_core_types - non_core_counts.len() + 1) as f32;
```

Comment:
Yeah I agree. This looks wrong to me. I'll fix this and make sure there's a candidate scoring test.

Reply:
Ok, I see what the issue was. There is a scoring test, and this is equivalent to the old version, but it's  redundant because, as you pointed out the score contribution for node types that are missing is 0. The thing that confused me here is that without this line, the num_non_core_types attribute is unused. 

For now I've removed the redundant lines and will clean up the warning in a separate commit.